driver/src/main/com/mongodb/client/gridfs/GridFSDownloadStreamImpl.java

```diff
@@ -155,11 +169,29 @@ private void checkClosed() {
         }
     }
 
-    private Document getChunk(final int chunkIndexToFetch) {
-        return chunksCollection.find(new Document("files_id", fileId).append("n", chunkIndexToFetch)).first();
+    private Document getChunk(final int startChunkIndex) {
+        if (cursor == null) {
+            cursor = chunksCollection.find(new Document("files_id", fileId).append("n", new Document("$gte", startChunkIndex)))
+                    .batchSize(batchSize).sort(new Document("n", 1)).iterator();
+        }
+        Document chunk = null;
+        if (cursor.hasNext()) {
+            chunk = cursor.next();
+            if (batchSize == 1) {
```

Comment:
> Why the check for batch size here?
Yes, it is a PITA - currently as the cursor closes, if I want to minimize memory use I have to get a new cursor each time (essentially its the same as the original way of doing `find().first()`). 
> Should it be responsible for validating that the "n" field equals chunkIndexToFetch?
I DRY'd it up but have copied it back.